cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -217,13 +217,33 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration) []string {
 			defaultArguments["audit-log-maxage"] = fmt.Sprintf("%d", *cfg.AuditPolicyConfiguration.LogMaxAge)
 		}
 	}
-
+	if cfg.APIServerExtraArgs == nil {
+		cfg.APIServerExtraArgs = map[string]string{}
+	}
+	cfg.APIServerExtraArgs["authorization-mode"] = getAuthzModes(cfg)
 	command = append(command, kubeadmutil.BuildArgumentListFromMap(defaultArguments, cfg.APIServerExtraArgs)...)
-	command = append(command, getAuthzParameters(cfg.AuthorizationModes)...)
 
 	return command
 }
 
+// getAuthzModes gets the authorization-related parameters to the api server
+// At this point, we can assume the list of authorization modes is valid (due to that it has been validated in the API
+// machinery code already) If the list is empty; it's defaulted (mostly for unit testing)
+func getAuthzModes(cfg *kubeadmapi.MasterConfiguration) string {
+	modeSet := sets.NewString()
+	defModes := strings.Split(kubeadmapiv1alpha2.DefaultAuthorizationModes, ",")
```

Comment:
I would build the logic like this:
1. Add everything you find in `extraArgsModes` splitted by the comma to a slice (e.g. `modeSlice`). Add the contents of that slice to a set.
2. Check whether `modeSet.Has("RBAC")`, and if not do `modeSlice = append([]string{"RBAC"}, modeSlice)`
3. Do the same for `"Node"`. Important is that Node must always be the first in the chain if specified, hence it needs to be prepended lastly